org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractHoverTest.xtend

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.testing
+
+import com.google.inject.Inject
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.jface.internal.text.html.BrowserInformationControlInput
+import org.eclipse.jface.text.IRegion
+import org.eclipse.jface.text.ITextHoverExtension2
+import org.eclipse.jface.text.Region
+import org.eclipse.xtext.resource.FileExtensionProvider
+import org.eclipse.xtext.ui.XtextProjectHelper
+import org.eclipse.xtext.ui.editor.XtextEditor
+import org.eclipse.xtext.ui.editor.XtextEditorInfo
+import org.eclipse.xtext.ui.editor.hover.IEObjectHover
+import org.eclipse.xtext.ui.refactoring.ui.SyncUtil
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil
+
+import static extension org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil.addNature
+
+/**
+ * @author miklossy - Initial contribution and API
+ * 
+ * @since 2.16
+ */
+abstract class AbstractHoverTest extends AbstractEditorTest {
```

Comment:
do you think it would be a good idea to add a sample usage to the javadoc
(maybe i am too much on a golang trip lately ;))

Reply:
I think coding sample usage in the javadoc does not make sense. The test cases added to the Xtext Example Projects should be efficient.